nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/server/ZooKeeperStateServer.java

```diff
@@ -136,7 +167,15 @@ private void startDistributed() throws IOException {
             quorumPeer.setInitLimit(quorumPeerConfig.getInitLimit());
             quorumPeer.setSyncLimit(quorumPeerConfig.getSyncLimit());
             quorumPeer.setQuorumVerifier(quorumPeerConfig.getQuorumVerifier(), false);
-            quorumPeer.setCnxnFactory(connectionFactory);
+
+            if (connectionFactory != null) {
+                quorumPeer.setCnxnFactory(connectionFactory);
+            }
+
+            if (secureConnectionFactory != null) {
+                quorumPeer.setCnxnFactory(secureConnectionFactory);
```

Comment:
Did you mean `quorumPeer.setSecureCnxnFactory(secureConnectionFactory)`?

Reply:
Yup. Funny, since the standalone state server calls `.startup()` / `.shutdown()` directly on the connection factories this never surfaced.